app/integration_action.go

```diff
@@ -141,15 +144,18 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 
 	if response.Update != nil {
 		response.Update.Id = postId
-		response.Update.AddProp("from_webhook", "true")
-		for key, value := range retain {
-			response.Update.AddProp(key, value)
-		}
-		for _, key := range remove {
-			delete(response.Update.Props, key)
-		}
-		if _, appErr = a.UpdatePost(response.Update, false); appErr != nil {
-			return "", appErr
+		if response.Update.Props == nil {
```

Comment:
Just to be clear: This is a breaking change for integrations that set `Update.Props` to `nil` to clear them. Is this correct?

Reply:
Affirmative.

Reply:
Integrations affected by this change can set `Update.Props` to `make(map[string]interface{})` to get the old behaviour?

Asking because this need to be remarked in the changelog.

Reply:
Right. 
1) update.Props == nil - carry over the old Props, entirely
2) update.Props == {} - clear all props, except the protected ones
3) update.Props == some_props - Post will be updated to "some_props", except the protected props

Reply:
Can we go ahead make a breaking change in a minor release though without a good reason? Can we work around that and break it with the 6.0 release, considering the feature is stable and customer integrations might break?

Reply:
@DSchalla I see you point. But I can't think of a work around for this.

To me this is one of the cases where you can argue that the old behaviour was a flaw and this is a "bug fix breaking change".

Reply:
1/5 agree with @hanzei. But I can totally see it the other way too.  Fixing up the other metadata (`IsPinned,HasReactions`) is what might have prompted the issue, I think we all agree that returning nil values from the action should not zero out prior values. Props should arguably be consistent, no?

Somewhat OT: personally I think we should be encouraging the explicit use of APIs (RPC or HTTP) rather than limited return values with special semantics. The only downside I know of using the APIs is a degradation in performance/latency, but I think it can be mitigated. The upsides are numerous, separate discussion.

Reply:
To confirm,

1 - there is no other way to prevent losing pinned message label and emoji reactions when the post is updated? We *must* do this breaking change?
2 - apart from our team, has anyone reported this flaw?
3 - what is the consequence for removing this setting for existing integrations?

Reply:
1. This change only deals with the problem of loosing `Props` when a post gets updated. Pinned messages and emoji reactions will be fixed with a different PR (or a different commit).
2. Yes, see https://github.com/matterpoll/matterpoll/issues/130
3. If integrations used `Update.Props == nil` to clear `Props`, this will no longer work.

Does this answer your questions?

Reply:
1 - Okay - in that case, is there no other way to prevent losing `Props` when the post is updated? We **must** do this breaking change?
2 - Okay, good to know 
3 - In which cases would integrations use `Update.Props == nil` to clear `Props`? (I realize I should know this, but I'm not sure)

Reply:
1. I can't think of another way. But maybe @levb has an idea.
3. I don't know. As this removes all `Props` this used may to much force. Using `Update.Props == nil` to clear `Props` seams to be more of a "wrong use" of the integration system.